Sources/FCM/Helpers/FCM+BatchSend.swift

```diff
@@ -2,40 +2,39 @@ import Foundation
 import Vapor
 
 extension FCM {
-    public func batchSend(_ message: FCMMessageDefault, tokens: String...) -> EventLoopFuture<[String]> {
-        _send(message, tokens: tokens)
+    public func batchSend(_ message: FCMMessageDefault, tokens: String...) async throws -> [String] {
+        try await _send(message, tokens: tokens)
     }
 
-    public func batchSend(_ message: FCMMessageDefault, tokens: String..., on eventLoop: EventLoop) -> EventLoopFuture<[String]> {
-        _send(message, tokens: tokens).hop(to: eventLoop)
+    public func batchSend(_ message: FCMMessageDefault, tokens: [String]) async throws -> [String] {
+        try await _send(message, tokens: tokens)
     }
 
-    public func batchSend(_ message: FCMMessageDefault, tokens: [String]) -> EventLoopFuture<[String]> {
-        _send(message, tokens: tokens)
-    }
-
-    public func batchSend(_ message: FCMMessageDefault, tokens: [String], on eventLoop: EventLoop) -> EventLoopFuture<[String]> {
-        _send(message, tokens: tokens).hop(to: eventLoop)
-    }
-
-    private func _send(_ message: FCMMessageDefault, tokens: [String]) -> EventLoopFuture<[String]> {
-        guard let configuration = self.configuration else {
-            fatalError("FCM not configured. Use app.fcm.configuration = ...")
-        }
-
+    private func _send(_ message: FCMMessageDefault, tokens: [String]) async throws -> [String] {
         let urlPath = URI(string: actionsBaseURL + configuration.projectId + "/messages:send").path
+        let accessToken = try await getAccessToken()
+
+        let chunks = tokens.chunked(into: 500)
+
+        return try await withThrowingTaskGroup(of: [String].self) { group in
```

Comment:
might make sense to execute those batches in series (they're chunked into batches for a reason) to not hit a throttled API (quadratic falloff etc.)

but fine for now, as this keeps the behavior as before 👌